ci/bamboo/build-windows.ps1

```diff
@@ -0,0 +1,227 @@
+copy "C:\Program Files\Git\usr\bin\msys*.dll" "C:\Program Files\PatchFromGit"
+$env:PATH = 'C:\Program Files\PatchFromGit;' + $env:PATH
+
+# Verify that patch is now available
+WrapCmd { where.exe patch }
+
+
+#
+# Remove sh.exe from the paths (CMake doesn't like it and fails in a subtle way)
+#
+
+# Validate expectation that sh.exe is in PATH prior to its removal
+WrapCmd { where.exe sh }
+
+$env:PATH = $env:PATH.Replace('C:\Program Files\OpenSSH\bin','')
+
+# Verify that sh command was successfully removed from PATH"
+&where.exe sh
+if ($LastExitCode -eq 0) {
+  throw "Failed to remove sh.exe from PATH."
+}
+
+
+# Verify that core toolchain components are available and log their versions
+Write-Host "Checking tools."
+WrapCmd { gcc --version }
+WrapCmd { g++ --version }
+
+WrapCmd { where.exe python }
+WrapCmd { python --version }
+Write-Host "PYTHONHOME=$env:PYTHONHOME"
+
+WrapCmd { pip --version }
+
+WrapCmd { wheel version }
+
+# Log installed python packages
+WrapCmd { pip list }
+
+
+#
+# Setup MinGW GCC as a valid distutils compiler
+#
+copy ".\external\windows64-gcc\bin\distutils.cfg" `
+     "$($env:PYTHONHOME)\Lib\distutils\distutils.cfg"
+
+
+#
+# Build nupic.core
+#
+
+mkdir .\build\release
+mkdir .\build\scripts
+
+pushd .\build\scripts
+
+$env:CC = "gcc"
+$env:CXX = "g++"
+
+WrapCmd {
+  cmake `
+    -G "MinGW Makefiles"  `
+    -DCMAKE_BUILD_TYPE="Release" `
+    -DCMAKE_INSTALL_PREFIX:PATH="..\release" `
+    -DNUPIC_BUILD_PYEXT_MODULES="ON" `
+    -DPY_EXTENSIONS_DIR:PATH="..\..\bindings\py\nupic\bindings" `
+    "..\.."
+}
+
+# Make nupic.core from non-debug configuration
+WrapCmd { cmake --build . --target install --config Release }
+
+popd
+
+
+# Create a python wheel in the destination wheelhouse
+Write-Host "Building nupic.bindings python wheel."
+WrapCmd {
+  python setup.py bdist_wheel `
+    --dist-dir "$($NupicCoreRootDir)\nupic_bindings_wheelhouse"
+}
+
+
+#
+# Run tests
+#
+
+# Install nupic.bindings before running c++ tests; py_region_test depends on it
+Write-Host "Installing from built nupic.bindings wheel."
+
+dir -Filter *.whl -Recurse | Select Fullname
+
+# Get path of nupic.bindings wheel
+$NupicBindingsWheel = `
+  (Get-ChildItem .\nupic_bindings_wheelhouse\nupic.bindings-*.whl)[0].FullName
+
+WrapCmd { pip install $NupicBindingsWheel }
+
+
+Write-Host "Running nupic.core C++ tests."
+
+pushd .\build\release\bin
+
+# TODO py_region_test.exe fails on windows;
+# https://github.com/numenta/nupic.core/issues/1075
+#WrapCmd { .\py_region_test.exe }
+
+WrapCmd { .\connections_performance_test.exe }
+WrapCmd { .\cpp_region_test.exe }
+WrapCmd { .\helloregion.exe }
+WrapCmd { .\hello_sp_tp.exe }
+WrapCmd { .\prototest.exe }
+WrapCmd { .\unit_tests.exe }
+popd
+
+
+Write-Host "Running nupic.bindings python tests."
+
+# So that py.test will deposit its artifacts in test_results
+mkdir .\test_results
+pushd .\test_results
+
+WrapCmd { python ..\setup.py test }
```

Comment:
@scottpurdy, regarding:
```
WrapCmd { python ..\setup.py test }
```
> The default behavior of this is to write results out to disk?
 Add a line note


It turned out that `python ..\setup.py test` is poorly behaved with respect to allowing control over the destination of the test artifacts, because it switches current working directory. This is done in tow places: 1. in root setup.py; and 2. in bindings/py/setup.py.

I ended up changing this to match the linux and os x nupic.bindings build script, which uses py.test to exercise control over the location of test artifacts.